src/tm-threads.c

```diff
@@ -362,6 +358,15 @@ static void *TmThreadsSlotPktAcqLoop(void *td)
     return NULL;
 }
 
+void TmThreadsWaitForUnpause(ThreadVars *tv)
+{
+    if (TmThreadsCheckFlag(tv, THV_PAUSE)) {
+        TmThreadsSetFlag(tv, THV_PAUSED);
+        TmThreadTestThreadUnPaused(tv);
+        TmThreadsUnsetFlag(tv, THV_PAUSED);
+    }
+}
```

Comment:
As this is now the only user of `TmThreadTestThreadUnPaused`, perhaps they should just be merged and this could be the new `TmThreadTestThreadUnPaused` - however, I feel the "Wait" in the name is more descriptive than Test, as `TmThreadTestThreadUnPaused` does wait as well.